pkg/kv/kvserver/client_merge_test.go

```diff
@@ -5266,14 +5266,14 @@ func TestStoreMergeGCHint(t *testing.T) {
 			dataLeft:   true,
 			dataRight:  true,
 			delLeft:    true,
-			expectHint: false,
+			expectHint: true,
```

Comment:
We should probably extend this test to assert what it originally wanted to assert, i.e. the `LatestRangeDeleteTimestamp` hint. We can add an additional assertion that it propagates the `GCTimestamp` hint.

Reply:
Done.